integrations/server_test.go

```diff
@@ -0,0 +1,519 @@
+						Name:     "billibob",
+						Provider: "github",
+						Scheme:   "oauth2",
+						Roles: []chronograf.Role{
+							{
+								Name:         "viewer",
+								Organization: "0",
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				server: &server.Server{
+					GithubClientID:     "not empty",
+					GithubClientSecret: "not empty",
+				},
+				method: "GET",
+				path:   "/chronograf/v1/users",
+				principal: oauth2.Principal{
+					Organization: "0",
+					Subject:      "billibob",
+					Issuer:       "github",
+					ExpiresAt:    time.Now().Add(10 * time.Second),
+					IssuedAt:     time.Now(),
+				},
+			},
+			wants: wants{
+				statusCode: 401,
+				body:       `{"code":401,"message":"User is not authorized"}`,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		testName := fmt.Sprintf("%s: %s", tt.name, tt.subName)
+		t.Run(testName, func(t *testing.T) {
+			ctx := context.TODO()
+			// Create Test Server
+			host, port := hostAndPort()
+			tt.args.server.Host = host
+			tt.args.server.Port = port
+
+			// This is so that we can use staticly generate jwts
+			tt.args.server.TokenSecret = "secret"
+
+			boltFile := newBoltFile()
+			tt.args.server.BoltPath = boltFile
+
+			// Prepopulate BoltDB Database for Server
+			boltdb := bolt.NewClient()
+			boltdb.Path = boltFile
+			_ = boltdb.Open(ctx)
+
+			// Populate Organizations
+			for i, organization := range tt.fields.Organizations {
+				o, err := boltdb.OrganizationsStore.Add(ctx, &organization)
+				if err != nil {
+					t.Fatalf("failed to add organization: %v", err)
+					return
+				}
+				tt.fields.Organizations[i] = *o
+			}
+
+			// Populate Users
+			for i, user := range tt.fields.Users {
+				u, err := boltdb.UsersStore.Add(ctx, &user)
+				if err != nil {
+					t.Fatalf("failed to add user: %v", err)
+					return
+				}
+				tt.fields.Users[i] = *u
+			}
+
+			// Populate Sources
+			for i, source := range tt.fields.Sources {
+				s, err := boltdb.SourcesStore.Add(ctx, source)
+				if err != nil {
+					t.Fatalf("failed to add source: %v", err)
+					return
+				}
+				tt.fields.Sources[i] = s
+			}
+
+			// Populate Servers
+			for i, server := range tt.fields.Servers {
+				s, err := boltdb.ServersStore.Add(ctx, server)
+				if err != nil {
+					t.Fatalf("failed to add server: %v", err)
+					return
+				}
+				tt.fields.Servers[i] = s
+			}
+
+			// Populate Layouts
+			for i, layout := range tt.fields.Layouts {
+				l, err := boltdb.LayoutsStore.Add(ctx, layout)
+				if err != nil {
+					t.Fatalf("failed to add layout: %v", err)
+					return
+				}
+				tt.fields.Layouts[i] = l
+			}
+
+			// Populate Dashboards
+			for i, dashboard := range tt.fields.Dashboards {
+				d, err := boltdb.DashboardsStore.Add(ctx, dashboard)
+				if err != nil {
+					t.Fatalf("failed to add dashboard: %v", err)
+					return
+				}
+				tt.fields.Dashboards[i] = d
+			}
+
+			_ = boltdb.Close()
+
+			go tt.args.server.Serve(ctx)
```

Comment:
Do you need to add a sync wait here?

Reply:
Would doing

```
var wg sync.WaitGroup
wg.Add(1)
go func() {
wg.Done()
tt.args.server.Serve(ctx)
}()

wg.Wait()
...
```

be
sufficient to fix the race? My thinking was that the race would still be there because it takes `Serve` some time to spin up.

My current work around was to just keep trying to hit an endpoint until it works.

Reply:
Thoughts here, @goller ^^ ?

Reply:
@desa yes

Reply:
@goller Just gave this a try and I'm still having issues. The only other way I can think of to solve this issue would be to have the server send a signal when it starts listening. Or possibly to add a `Listening` method to `Server` that reports whether the server is listening. Or I could leave it as it is now. Which do you think would be best?